### ISSUE TITLE
no-unnecessary-type-annotation: false positive with array of tuples

### ISSUE THREAD

**Issue by ajafff** [ajafff/tslint-consistent-codestyle#46]
```ts
let foo: Array<[string, boolean]> = [
  ['a', true],
  ['b', false],
];
```
